car-test.mjs

```diff
@@ -60,21 +60,21 @@ async function run () {
         Readable.from(out).pipe(fs.createWriteStream(filename))
       }
 
-      writer.put({ cid: cidA, bytes: bytesA })
+      await writer.put({ cid: cidA, bytes: bytesA })
```

Comment:
`await`ing on `writer.put()` is not critical, it will batch up the writes and handle them in the background, you just have to be OK with the memory build-up that's going to incur. We made sure to support the write,write,write,...,await-close model for dumbo-drop so we could parallelise, there's a memory cost but maybe a slight perf boost.

Reply:
When I try to run your code with the extra `await` here, the Node.js process exits (no exception or anything) ... there must be some event loop magic I don't understand.